ThingIFSDK/ThingIFSDK/Gateway/GatewayAPI.swift

```diff
@@ -67,13 +67,13 @@ public class GatewayAPI: NSObject, NSCoding {
             return
         }
 
-        let requestURL = "\(self.app.baseURL)/\(self.app.siteName)/token"
+        let requestURL = "\(self.gatewayAddress.absoluteString)/\(self.app.siteName)/token"
 
         // generate header
         let credential = "\(self.app.appID):\(self.app.appKey)"
         let base64Str = credential.dataUsingEncoding(NSUTF8StringEncoding)?.base64EncodedDataWithOptions(NSDataBase64EncodingOptions.Encoding64CharacterLineLength)
```

Comment:
@satoshikumano 
```
[NSData base64EncodedStringWithOptions]
```
is used for decoding base64 string to plain nsdata.
In this case, we are encoding plain string into base 64 string.
I will rename the variable name to make it clear.